app-backend/db/src/main/scala/AnnotationProjectDao.scala

```diff
@@ -79,6 +158,147 @@ object AnnotationProjectDao
+    (selectF ++ Fragments.whereAndOpt(Some(fr"id = ${id}")))
+      .query[AnnotationProject]
+  }
+
+  def getById(id: UUID): ConnectionIO[Option[AnnotationProject]] =
+    annotationProjectByIdQuery(id).option
+
+  def getWithRelatedById(
+      id: UUID
+  ): ConnectionIO[Option[AnnotationProject.WithRelated]] =
+    for {
+      projectO <- getById(id)
+      tileLayers <- TileLayerDao.listByProjectId(id)
+      labelClassGroup <- AnnotationLabelClassGroupDao
+        .listByProjectId(id)
+      labelClassGroupWithClass <- labelClassGroup traverse { group =>
+        AnnotationLabelClassDao
+          .listAnnotationLabelClassByGroupId(group.id)
+          .map(group.withLabelClasses(_))
+      }
+    } yield {
+      projectO map { project =>
+        project.withRelated(tileLayers, labelClassGroupWithClass)
+      }
+    }
+
+  def getWithRelatedAndSummaryById(
+      id: UUID
+  ): ConnectionIO[Option[AnnotationProject.WithRelatedAndSummary]] =
+    for {
+      withRelatedO <- getWithRelatedById(id)
+      taskStatusSummary <- TaskDao.countProjectTaskByStatus(id)
+      labelClassGroups <- AnnotationLabelClassGroupDao.listByProjectId(id)
+      labelClassSummaries <- labelClassGroups traverse { labelClassGroup =>
+        AnnotationLabelDao.countByProjectAndGroup(id, labelClassGroup.id).map {
+          summary =>
+            AnnotationProject.LabelClassGroupSummary(
+              labelClassGroup.id,
+              labelClassGroup.name,
+              summary
+            )
+        }
+      }
+    } yield {
+      withRelatedO map { withRelated =>
+        withRelated.withSummary(
+          taskStatusSummary,
+          labelClassSummaries
+        )
+      }
+    }
+
+  def deleteById(id: UUID): ConnectionIO[Int] =
+    query.filter(fr"id = ${id}").delete
+
+  def update(project: AnnotationProject, id: UUID): ConnectionIO[Int] = {
+    (fr"UPDATE " ++ tableF ++ fr"""SET
+      name = ${project.name},
+      labelers_team_id = ${project.labelersTeamId},
+      validators_team_id = ${project.validatorsTeamId},
+      "task_size_meters"= ${project.taskSizeMeters},
+      "aoi" = ${project.aoi}
+    WHERE
+      id = $id
+    """).update.run;
+  }
+
+  def updateWithRelated(
+      id: UUID,
+      projectWithRelated: AnnotationProject.WithRelated
+  ): ConnectionIO[Int] = {
+    val project = projectWithRelated.toProject
+    for {
+      count <- update(project, id)
+      _ <- TileLayerDao.deleteByProjectId(id)
+      _ <- projectWithRelated.tileLayers traverse { layer =>
+        TileLayerDao.insertTileLayer(
+          TileLayer.Create(
+            layer.name,
+            layer.url,
+            Some(layer.default),
+            Some(layer.overlay),
+            layer.layerType
+          ),
+          project
+        )
+      }
+      _ <- AnnotationLabelClassGroupDao.deleteByProjectId(id)
+      _ <- projectWithRelated.labelClassGroups.zipWithIndex traverse {
```

Comment:
Wait actually... I didn't understand this before. It's super unsafe to update a project this way, since deleting the old label class groups deletes the old label classes... which means that any labels associated with those classes will also die. I didn't have any labels so everything was fine, but this makes simple project update into a fairly intense db operation that destroys data.